app/controllers/reviews_controller.rb

```diff
@@ -0,0 +1,57 @@
+class  ReviewsController < ApplicationController
+
+  def new
+    @review = Review.new
+  end
+
+  def create
+    @buddy = Buddy.find(params[:buddy_id])
+    @review = Review.new(review_params)
+    @review.buddy_id = @buddy.id
+    @review.user_id = current_user.id
+    if @review.save
+      flash[:notice] = "You've successfully submitted a review!"
+      redirect_to buddy_path(@buddy)
+    else
+      flash[:alert] = @review.errors.full_messages
+      redirect_to buddy_path(@buddy)
```

Comment:
I would `render :new` here instead of redirecting. This will preserve any information that is stored in the `@review` model which pre-populates the form (i.e. the user won't lose the review they just typed out).